pinc/graph_data.inc

```diff
@@ -289,3 +289,212 @@ function users_by_country()
         ],
     ];
 }
+
+function array_successive_differences($arr)
+// Return the array defined by:
+//     $result[$i] = $arr[$i+1] - $arr[$i]
+{
+    $result = [];
+    foreach ($arr as $key => $value) {
+        if (isset($prev_key)) {
+            $result[$prev_key] = $value - $prev_value;
+        }
+        $prev_key = $key;
+        $prev_value = $value;
+    }
+    return $result;
+}
+
+function get_number_of_days_in_current_month()
+{
+    $current = getdate();
+    // The last day of this month is the "zeroth" day of next month.
+    $last_day_timestamp =
+        mktime(0, 0, 0, $current['mon'] + 1, 0, $current['year']);
+    $last_day = getdate($last_day_timestamp);
+    return $last_day['mday'];
+}
+
+function curr_month_proj($which)
+{
+    // Create "projects Xed per day" graph for current month
+
+    $psd = get_project_status_descriptor($which);
+
+    $todaysTimeStamp = time();
+
+    $year = date("Y", $todaysTimeStamp);
+    $month = date("m", $todaysTimeStamp);
+    $monthVar = _(date("F", $todaysTimeStamp));
+    $timeframe = "$monthVar $year";
+
+    $maxday = get_number_of_days_in_current_month();
+
+    //query db and put results into arrays
+    $sql = sprintf("SELECT DAYOFMONTH(date) as day, SUM(num_projects)
+      FROM project_state_stats
```

Comment:
I guess the formatter isn't enforcing the indents of these strings? Please indent these 4 spaces. Ditto the other SQL in the subsequent functions (some of which are indented 1 space from the parent, and at least one with 0 indents).

Reply:
Is it supposed to enforce code indenting inside strings? If not, then we'll need to be more conscious about remembering to do it manually for this situation.